packages/webvh/src/anoncreds/services/WebVhAnonCredsRegistry.ts

```diff
@@ -95,65 +94,55 @@ export class WebVhAnonCredsRegistry implements AnonCredsRegistry {
-      if (!isProofValid) {
+      // 2. DataIntegrityProof Verification
+      const verified = await this.verifyProof(agentContext, resourceObject)
+      if (!verified) {
+        agentContext.config.logger.trace('Resolved resource proof is invalid.')
         throw new CredoError('Resolved resource proof is invalid.')
       }
       agentContext.config.logger.trace(`Resource ${resourceId} proof validated.`)
 
-      // 3. Check filename hash against content hash
-      const contentObject = resourceObject.content
-      if (!contentObject) {
-        throw new CredoError('Resolved resource data does not contain a content object.')
+      // 3. Resource Digest Multibase Verification
+      if (resourceObject.id !== resourceId) {
+        throw new CredoError(`ID mismatch ${resourceObject.id} != ${resourceId}`)
       }
-
       const parts = resourceId.split(/[:/]/)
       const expectedMultibaseHash = parts[parts.length - 1]
       if (!expectedMultibaseHash) {
         throw new CredoError(`Could not extract expected hash from ${resourceTypeString} ID: ${resourceId}`)
       }
+      const digestMultibase = this._digestMultibase(canonicalize(resourceObject.content))
 
-      const contentString = canonicalize(contentObject)
-      const actualHashMultibase = MultiBaseEncoder.encode(
-        MultiHashEncoder.encode(TypedArrayEncoder.fromString(contentString), 'sha-256'),
-        'base58btc'
-      )
-
-      if (actualHashMultibase !== expectedMultibaseHash) {
+      if (digestMultibase !== expectedMultibaseHash) {
         throw new CredoError(
-          `Content hash mismatch for ${resourceId}. Expected: ${expectedMultibaseHash}, Actual: ${actualHashMultibase}`
+          `Content hash mismatch for ${resourceId}. Expected: ${expectedMultibaseHash}, Actual: ${digestMultibase}`
         )
       }
       agentContext.config.logger.trace(`Resource ${resourceId} content hash matches filename.`)
 
-      // 4. Check that the issuerId in the content matches the DID part of the resourceId
-      const contentIssuerId = (contentObject as { issuerId?: string })?.issuerId // Type assertion for accessing issuerId
-      if (!contentIssuerId || typeof contentIssuerId !== 'string') {
-        throw new CredoError(`Resolved resource content for ${resourceId} is missing a valid issuerId.`)
-      }
-
-      // Extract the DID from the resourceId (part before /resources/hash)
-      const resourceDidMatch = resourceId.match(/^(did:webvh:[^/]+)/)
-      if (!resourceDidMatch || !resourceDidMatch[1]) {
-        throw new CredoError(`Could not extract DID from resource ID: ${resourceId}`)
-      }
-      const expectedIssuerDid = resourceDidMatch[1]
-
-      if (contentIssuerId !== expectedIssuerDid) {
-        throw new CredoError(
-          `Issuer ID mismatch for ${resourceId}. Expected: ${expectedIssuerDid}, Actual: ${contentIssuerId}`
-        )
-      }
-      agentContext.config.logger.trace(`Resource ${resourceId} issuer ID matches DID.`)
```

Comment:
Why is this check removed?

Reply:
I suppose we can keep it, I removed it because not all attested resource will have a `content.issuerId` property